OMPython/__init__.py

```diff
@@ -238,8 +239,11 @@ def sendExpression(self, command):
             self._omc = None
             return result
           else:
-            answer = OMTypedParser.parseString(result)
-            return answer
+            try:
+              answer = OMTypedParser.parseString(result)
+              return answer
+            except:
```

Comment:
If you look into the code, the results of the compiler is returned not the
string we wanted to parse, because the Pyparsing does not give proper error
messages in case there are errors in the code

regards
arun

On Fri, Apr 29, 2016 at 2:37 PM, Christoph Buchner <notifications@github.com
> wrote:

> In OMPython/__init__.py
> <https://github.com/OpenModelica/OMPython/pull/10#discussion_r61570107>:
>
> > @@ -238,8 +239,11 @@ def sendExpression(self, command):
> >              self._omc = None
> >              return result
> >            else:
> > -            answer = OMTypedParser.parseString(result)
> > -            return answer
> > +            try:
> > +              answer = OMTypedParser.parseString(result)
> > +              return answer
> > +            except:
>
>
>    1. It's typically bad python style to catch all exceptions without
>    qualification, you should probably limit this to PyPArsingExceptions or
>    whatever the type of the exceptions you expect are.
>    2. In the case of the exception, shouldn't you return some info about
>    the exception, e.g. its message? You just return the string you wanted to
>    parse in the first place.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenModelica/OMPython/pull/10/files/8c0f91d26d68440d1fe8c8825124e58cdf39100c#r61570107>
>